src/components/Avatar/Avatar.svelte

```diff
@@ -0,0 +1,85 @@
+<script lang="ts">
+    import type { Action } from "svelte/action";
+    
+    // Note: consider `./types.ts`
+    interface AvatarProps {
+        // initials
+        initials?: string;
+        initialsFill?: string;
+        initialsFontSize?: number;
+        initialsBase?: string;
+        initialsRest?: string;
+
+        // image
+        imageBase?:string;
+        imageSrc?: string;
+        imageAlt?: string;
+        imageFallback?: string;
+        imageAction?: Action<HTMLElement, string>;
+        imageActionParams?: string;
+        imageRest?: string;
+
+        // figure
+        figureBase?: string;
```

Comment:
While `figure` is semantically correct, we need to inform users which is the root/parent element. In this case switch from `figureX` to `rootX`.

Plus `rootX` props should be closer to the top, to match their position in the template. Just below functional props like `action/actionParams`